### PR TITLE
Add rubygem-smart_proxy_dhcp_remote_isc package

### PR DESCRIPTION
**Pull request by witlessbird**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.14
* [ ] 1.13
* [ ] 1.12

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+Name: rubygem-%{gem_name}
+Version: 0.0.2
+Release: 1%{?foremandist}%{?dist}
+Summary: Smart-Proxy dhcp module provider for NFS-accessible ISC dhcpd installations
+Group: Applications/Internet
+License: GPLv3
+URL: https://github.com/theforeman/smart_proxy_dhcp_remote_isc
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: foreman-proxy >= 1.15.0
+%if 0%{?rhel} == 6
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: This conditional's obsolete, can be removed, only needs `ruby(release)`.

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Requires: ruby(abi)
+%else
+Requires: ruby(release)
+%endif
+Requires: ruby
+Requires: ruby(rubygems)
+%if 0%{?rhel} == 6
+BuildRequires: ruby(abi)
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: ditto, remove this part of the conditional

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# sample config
+mkdir -p %{buildroot}%{foreman_proxy_settingsd_dir}
+mv %{buildroot}%{gem_instdir}/config/dhcp_remote_isc.yml.example \
+   %{buildroot}%{foreman_proxy_settingsd_dir}/dhcp_remote_isc.yml
+
+%files
+%dir %{gem_instdir}
+%config(noreplace) %attr(0640, root, foreman-proxy) %{foreman_proxy_settingsd_dir}/dhcp_remote_isc.yml
+%doc %{gem_instdir}/LICENSE
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: `%license`

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/README.md
+%{gem_instdir}/test
+
+%changelog
+* Wed Mar 22 2017 wb - 0.0.1-1
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: Please remove this, it's not needed (will be generated when built).

**Comment by witlessbird** [comment]
- removed rhel-6 conditionals
 - switched to %license macro
 - removed changelog

**Comment by domcleal** [comment]
Please also check the two comments at https://github.com/theforeman/foreman-packaging/pull/1577#pullrequestreview-29173558.

**Comment by witlessbird** [comment]
- added doc subpackages to comps
- added rubygems-smart_proxy_dhcp_remote_isc to foreman-plugins-nightly-nonscl-rhel7 in tito.props

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+Summary: Smart-Proxy dhcp module provider for NFS-accessible ISC dhcpd installations
+Group: Applications/Internet
+License: GPLv3
+URL: https://github.com/theforeman/smart_proxy_dhcp_remote_isc
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: foreman-proxy >= 1.15.0
+Requires: ruby(release)
+Requires: ruby
+Requires: ruby(rubygems)
+BuildRequires: rubygems-devel
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: Missing `BuildRequires: ruby(release)`, may have been removed when removing the EL6 conditional.

File: rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+%{gem_libdir}
+%{foreman_proxy_bundlerd_dir}/%{plugin_name}.rb
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/README.md
+%{gem_instdir}/test
+
```
Comment (domcleal) [inline rubygem-smart_proxy_dhcp_remote_isc/rubygem-smart_proxy_dhcp_remote_isc.spec]: Please add the `%changelog` header from the template, it was only the entry I meant to be removed.

**Comment by witlessbird** [comment]
- added ```BuildRequires: ruby(release)```
- added ```%changelog``` macro

**Comment by domcleal** [comment]
Merged and building, thanks @witlessbird.

**Pull request merged**